int32 id
UserJoint[] joints
int32 waitTime
